test/jdk/javax/sound/midi/File/SMFInterruptedRunningStatus.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.ByteArrayInputStream;
+
+import javax.sound.midi.MidiSystem;
+import javax.sound.midi.Sequence;
+import javax.sound.midi.Track;
+
+/**
+ * @test
+ * @bug 8319598
+ * @summary SMFParser bug with running status, interrupted by Meta or SysEx messages
+ */
+public class SMFInterruptedRunningStatus {
+
+    public static void main(String[] args) throws Exception {
+
+        byte[][] files = new byte[][] {SMF_1, SMF_2, SMF_3};
+        for (int i = 0; i < files.length; i++) {
+            Sequence seq = MidiSystem.getSequence(new ByteArrayInputStream(files[i]));
+            testSequence(seq, i + 1);
+        }
+
+        // no exception thrown, all files have been parsed correctly
+        System.out.println("Test passed");
+    }
+
+    private static void testSequence(Sequence seq, int fileNumber) {
+
+        // check number of tracks and number of events
+        Track[] tracks = seq.getTracks();
+        if (1 != tracks.length) {
+            throw new RuntimeException("file number " + fileNumber + " fails (incorrect number of tracks: " + tracks.length + ")");
+        }
+        Track track = tracks[0];
+        if (7 != track.size()) {
+            throw new RuntimeException("file number " + fileNumber + " fails (incorrect number of events: " + track.size() + ")");
+        }
+
+        // check status byte of each message
+        int[] expectedStatusBytes = new int[] {0x90, 0xFF, 0x90, 0x90, 0x90, 0xFF, 0xFF};
+        for (int i = 0; i < expectedStatusBytes.length; i++) {
+            int expected = expectedStatusBytes[i];
+            if (expected != track.get(i).getMessage().getStatus()) {
+                throw new RuntimeException("file number " + fileNumber + " fails (wrong status byte in event " + i + ")");
+            }
+        }
+    }
+
```

Comment:
please split the long line above to have 80 chars per line.

The code below for "SMF_1"/etc are fine as is.

Reply:
Thank you for the suggestion. I reduced the line lengths.
Also for the code below SMF_..., even if you think it was fine.
I think now it's even more readable. But I can redo this part, if you think it was better before.